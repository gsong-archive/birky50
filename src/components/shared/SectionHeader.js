//@flow strict
import * as React from "react";

import facepaint from "facepaint";
import styled, { css } from "react-emotion";

import SectionContext from "../../contexts/SectionContext";

import type { LabelComponent } from "./EmojiLabels";

type Props = { children: React.Element<LabelComponent> };

const SectionHeader = ({ children }: Props) => (
  <SectionContext.Consumer>
    {id => <H1 id={`${id}-description`}>{children}</H1>}
  </SectionContext.Consumer>
);

const breakpoints = [350];
const mq = facepaint(breakpoints.map(bp => `@media (max-width: ${bp}px)`));

const H1 = styled.h1(
  css`
    line-height: 1.1;
    margin-bottom: 1rem;
  `,
  mq({ fontSize: [null, "2.4rem"] })
);

export default SectionHeader;
