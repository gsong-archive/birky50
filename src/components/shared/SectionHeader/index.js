//@flow strict
import * as React from "react";

import facepaint from "facepaint";
import styled, { css } from "react-emotion";

import SectionContext from "../../../contexts/SectionContext";
import { convertListToEm } from "../../utils";

import type { LabelComponent } from "../EmojiLabels";

type Props = { children: React.Element<LabelComponent> };

const SectionHeader = ({ children }: Props) => (
  <SectionContext.Consumer>
    {id => <H1 id={`${id}-description`}>{children}</H1>}
  </SectionContext.Consumer>
);

const breakpoints = convertListToEm([350]);
const mq = facepaint(breakpoints.map(bp => `@media (max-width: ${bp}em)`));

const H1 = styled.h1(
  css`
    font-weight: normal;
    line-height: 1.1;
    margin: 0 0 1em;
  `,
  mq({ fontSize: ["1.75rem", "1.5rem"] })
);

export default SectionHeader;
