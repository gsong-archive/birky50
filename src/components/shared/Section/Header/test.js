import React from "react";

import { render } from "react-testing-library";

import SectionHeader from "./index";

test("SectionHeader is rendered", () => {
  const { container } = render(<SectionHeader>I'm a child.</SectionHeader>);

  expect(container.firstChild).toMatchSnapshot();
});
