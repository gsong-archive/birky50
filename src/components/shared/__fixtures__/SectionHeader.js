import React from "react";

import { generate } from "testUtils";

import SectionHeader from "../SectionHeader";
import { Airplane } from "../EmojiLabels";

export default [
  {
    name: "With EmojiLabel",
    component: SectionHeader,
    props: {
      children: <Airplane label={generate.words()} />,
    },
  },
  {
    name: "With text label",
    component: SectionHeader,
    props: {
      children: generate.words(),
    },
  },
];
