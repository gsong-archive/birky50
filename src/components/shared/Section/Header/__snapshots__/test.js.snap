// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHeader is rendered 1`] = `
.emotion-2 {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  background-color: hsla(0,0%,100%,0.8);
  padding: 15px 24px;
  border-top: 8px solid yellow;
}

@media (max-width:21.875em) {
  .emotion-2 {
    padding: 12px 24px;
  }
}

.emotion-0 {
  font-weight: normal;
  margin: 0 0;
  font-size: 1.75rem;
}

@media (max-width:21.875em) {
  .emotion-0 {
    font-size: 1.5rem;
  }
}

<div
  class="emotion-2 emotion-3"
>
  <h1
    id="uniqueId-description"
    class="emotion-0 emotion-1"
  >
    I'm a child.
  </h1>
</div>
`;
